controllers/vrg_volrep.go

```diff
@@ -508,17 +509,43 @@ func (v *VRGInstance) undoPVRetentionForPVC(pvc corev1.PersistentVolumeClaim, lo
 	return nil
 }
 
+func (v *VRGInstance) generateArchiveAnnotation(gen int64) string {
+	return fmt.Sprintf("%s-%s", pvcVRAnnotationArchivedVersionV1, strconv.Itoa(int(gen)))
+}
+
+func (v *VRGInstance) isArchivedAlready(pvc *corev1.PersistentVolumeClaim, log logr.Logger) bool {
+	pvHasAnnotation := false
+	pvcHasAnnotation := false
+
+	pv, err := v.getPVFromPVC(pvc)
+	if err != nil {
+		log.Error(err, "Failed to get PV from PVC, PVC:%v, err: %w", pvc.Name, err)
+
+		return false
+	}
+
+	pvcDesiredValue := v.generateArchiveAnnotation(pvc.Generation)
+	if v, ok := pvc.ObjectMeta.Annotations[pvcVRAnnotationArchivedKey]; ok && (v == pvcDesiredValue) {
```

Comment:
Using the `generation` as a suffix for the `archiveV1` will not work. The generation can change after failover/relocation

Reply:
I thought about it and although not ideal there is no better number than generation to track this. It will definitely result in one upload right after restore.